src/libraries/System.IO.Compression/src/System/IO/Compression/ZLibStream.cs

```diff
@@ -0,0 +1,293 @@
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.IO.Compression
+{
+    /// <summary>Provides methods and properties used to compress and decompress streams by using the ZLib data format specification.</summary>
+    public sealed class ZLibStream : Stream
+    {
+        /// <summary>The underlying deflate stream.</summary>
+        private DeflateStream _deflateStream;
+
+        /// <summary>Initializes a new instance of the <see cref="ZLibStream"/> class by using the specified stream and compression mode.</summary>
+        /// <param name="stream">The stream to which compressed data is written or from which decompressed data is read.</param>
+        /// <param name="mode">One of the enumeration values that indicates whether to compress or decompress the stream.</param>
+        public ZLibStream(Stream stream, CompressionMode mode) : this(stream, mode, leaveOpen: false)
+        {
+        }
+
+        /// <summary>Initializes a new instance of the <see cref="ZLibStream"/> class by using the specified stream, compression mode, and whether to leave the <paramref name="stream"/> open.</summary>
+        /// <param name="stream">The stream to which compressed data is written or from which decompressed data is read.</param>
+        /// <param name="mode">One of the enumeration values that indicates whether to compress or decompress the stream.</param>
+        /// <param name="leaveOpen"><c>true</c> to leave the stream object open after disposing the <see cref="ZLibStream"/> object; otherwise, <c>false</c>.</param>
+        public ZLibStream(Stream stream, CompressionMode mode, bool leaveOpen)
+        {
+            _deflateStream = new DeflateStream(stream, mode, leaveOpen, ZLibNative.ZLib_DefaultWindowBits);
+        }
+
+        /// <summary>Initializes a new instance of the <see cref="ZLibStream"/> class by using the specified stream and compression level.</summary>
+        /// <param name="stream">The stream to which compressed data is written.</param>
+        /// <param name="compressionLevel">One of the enumeration values that indicates whether to emphasize speed or compression efficiency when compressing the stream.</param>
+        public ZLibStream(Stream stream, CompressionLevel compressionLevel) : this(stream, compressionLevel, leaveOpen: false)
+        {
+        }
+
+        /// <summary>Initializes a new instance of the <see cref="ZLibStream"/> class by using the specified stream, compression level, and whether to leave the <paramref name="stream"/> open.</summary>
+        /// <param name="stream">The stream to which compressed data is written.</param>
+        /// <param name="compressionLevel">One of the enumeration values that indicates whether to emphasize speed or compression efficiency when compressing the stream.</param>
+        /// <param name="leaveOpen"><c>true</c> to leave the stream object open after disposing the <see cref="ZLibStream"/> object; otherwise, <c>false</c>.</param>
```

Comment:
```suggestion
        /// <param name="leaveOpen"><see langword="true" /> to leave the stream object open after disposing the <see cref="ZLibStream"/> object; otherwise, <see langword="false" />.</param>
```